### PR TITLE
update to 24.0

### PR DESCRIPTION
**Pull request by cbouss**
pip 24.0

**Destination channel:** main

### Links

- https://anaconda.atlassian.net/browse/PKG-3682
- https://github.com/pypa/pip/tree/24.0
- https://github.com/pypa/pip/blob/24.0/NEWS.rst

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,9 +37,34 @@ test:
 
 about:
   home: https://pip.pypa.io
-  license: MIT
   license_family: MIT
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml:40]: It accepts Other. I'll set that instead.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ package:
 
 source:
   url: https://pypi.io/packages/source/p/pip/pip-{{ version }}.tar.gz
-  sha256: 1fcaa041308d01f14575f6d0d2ea4b75a3e2871fe4f9c694976f908768e14174
+  sha256: ea9bd1a847e8c5774a5777bb398c19e80bcd4e2aa16a4b301b718fe6f593aba2
 
 build:
   number: 0
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml:14]: Addressed, thank you

File: recipe/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Addressed, thank you

File: recipe/meta.yaml [hunk 3]
```diff
@@ -37,9 +37,34 @@ test:
 
 about:
   home: https://pip.pypa.io
-  license: MIT
-  license_family: MIT
-  license_file: LICENSE.txt
+  license_family: Other
+  license: BSD-2-Clause AND LGPL-2.1 AND Apache-2.0 AND EPL-2.0 AND MIT AND BSD-3-Clause
+  license_file:
```
Comment (Jrice1317) [inline recipe/meta.yaml:42]: Not a blocker, but it would be nice if these were in alphabetical order, like in the repo. Lol Easier to track

Comment (Jrice1317) [reply recipe/meta.yaml:42]: Also, it seems we could add the following:
- src/pip/_vendor/six.LICENSE
- src/pip/_vendor/typing_extensions.LICENSE

**Review by Jrice1317 — COMMENTED** [review]
LGTM outside of the two missing licenses

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:42]: Thanks, addressed through new commit

**Review by danpetry — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -37,9 +37,36 @@ test:
 
 about:
   home: https://pip.pypa.io
-  license: MIT
-  license_family: MIT
-  license_file: LICENSE.txt
+  license_family: Other
+  license: BSD-2-Clause AND LGPL-2.1 AND Apache-2.0 AND EPL-2.0 AND MIT AND BSD-3-Clause
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:41]:
```suggestion
  license: BSD-2-Clause AND LGPL-2.1-or-later AND Apache-2.0 AND EPL-2.0 AND MIT AND BSD-3-Clause AND MPL-2.0 AND ISC
```
- `src/pip/_vendor/certifi/LICENSE` is [MPL-2.0](https://github.com/pypa/pip/blob/24.0/src/pip/_vendor/certifi/LICENSE#L16-L18)
- `src/pip/_vendor/chardet/LICENSE` is [LGPL-2.1-or-later](https://github.com/pypa/pip/blob/24.0/src/pip/_vendor/chardet/LICENSE)
- `src/pip/_vendor/resolvelib/LICENSE` is [ISC](https://github.com/pypa/pip/blob/24.0/src/pip/_vendor/resolvelib/LICENSE)

See also https://fossies.org/linux/misc/pip-24.0.tar.gz/meta.html

However, maintaining this list is quite burdensome

Comment (cbouss) [reply recipe/meta.yaml]: change reverted to avoid confusion

**Pull request merged**